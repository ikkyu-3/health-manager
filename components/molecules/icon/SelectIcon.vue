<template>
  <div class="icon" :class="classObject">
    <span v-if="isActive">{{ text }}</span>
  </div>
</template>

<script lang="ts">
import { createComponent, computed } from '@vue/composition-api'

export default createComponent({
  props: {
    text: {
      type: String,
      required: true,
      default: ''
    }
  },
  setup(props) {
    const isActive = computed(() => props.text.length !== 0)
    const classObject = computed(() => ({ active: isActive.value }))

    return { isActive, classObject }
  }
})
</script>

<style scoped>
.icon {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 32px;
  width: 32px;
  border-radius: 50%;
  background-color: #bdbdbd;
  color: #ffffff;
  font-weight: bold;
}

.active {
  background-color: #ef9a9a !important;
}
</style>
