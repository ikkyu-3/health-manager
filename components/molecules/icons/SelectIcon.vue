<template>
  <div class="icon" :class="classObject">
    <span v-if="isActive">{{ text }}</span>
  </div>
</template>

<script lang="ts">
import { createComponent, computed } from '@vue/composition-api'

export default createComponent({
  props: {
    text: {
      type: String,
      required: true,
      default: ''
    }
  },
  setup(props) {
    const isActive = computed(() => props.text.length !== 0)
    const classObject = computed(() => ({ active: isActive.value }))

    return { isActive, classObject }
  }
})
</script>

<style scoped lang="scss">
.icon {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 32px;
  width: 32px;
  border-radius: 50%;
  background-color: #bdbdbd;
  color: #fff;
  font-weight: bold;
}

.icon.active {
  background-color: #ef9a9a;
}
</style>
