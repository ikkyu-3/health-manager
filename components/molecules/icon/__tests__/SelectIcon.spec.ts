import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import SelectIcon from '@/components/molecules/icon/SelectIcon.vue'

const options = {
  localVue: createLocalVue(),
  vuetify: new Vuetify()
}

describe('molecules/icon/SelectIcon.vue', () => {
  it('textが空文字の場合、何も表示されないアイコンが表示される', () => {
    const wrapper = mount(SelectIcon, {
      ...options,
      propsData: { text: '' }
    })
    expect(wrapper.find('div').classes()).not.toEqual(
      expect.arrayContaining(['active'])
    )
    expect(wrapper.find('div').text()).toBe('')
  })

  it('textが文字列の場合、textの文字がアイコンに表示される', () => {
    const wrapper = mount(SelectIcon, {
      ...options,
      propsData: { text: '1' }
    })
    expect(wrapper.find('div').classes()).toEqual(
      expect.arrayContaining(['active'])
    )
    expect(wrapper.find('div').text()).toBe('1')
  })
})
