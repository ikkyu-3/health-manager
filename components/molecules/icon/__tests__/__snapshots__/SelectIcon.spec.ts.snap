// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/icon/SelectIcon.vue snapshot 1`] = `"<div class=\\"icon active\\"><span>1</span></div>"`;
